packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -3468,3 +3505,34 @@ class InputDecorationTheme extends Diagnosticable {
     properties.add(DiagnosticsProperty<bool>('alignLabelWithHint', alignLabelWithHint, defaultValue: defaultTheme.alignLabelWithHint));
   }
 }
+
+/// The vertical alignment of text within an input.
+///
+/// This consists simply of a double [y] that can range from -1.0 to 1.0. -1.0
+/// aligns to the top of the input so that the top of the first line of text
+/// fits within the input and its padding. 0.0 aligns to the center of the
+/// input. 1.0 aligns so that the bottom of the last line of text aligns with
+/// the bottom interior edge of the input.
+class TextAlignVertical {
+  /// Construct TextAlignVertical from any given y value.
+  const TextAlignVertical({
+    @required this.y,
+  }) : assert(y != null),
+       assert(y >= -1.0 && y <= 1.0);
+
+  /// A value ranging from -1.0 to 1.0 that corresponds to the topmost to the
```

Comment:
that corresponds to the topmost to the => bottommost possible text position => that defines the topmost and bottommost locations of the top and bottom of the input text box.